registry/txt.go

```diff
@@ -135,6 +144,18 @@ func (im *TXTRegistry) Records(ctx context.Context) ([]*endpoint.Endpoint, error
 				ep.Labels[k] = v
 			}
 		}
+		// Handle the migration of TXT records created before the new format (introduced in v0.12.0)
+		if len(txtRecordsMap) > 0 {
+			if isManagedRecord(ep, im.managedRecordTypes) {
```

Comment:
We noticed that in GCP, this winds up creating a TXT record for every A and CNAME DNS record in our DNS zones, whether or not those records are managed by ExternalDNS. If I'm reading the code correctly, the `(*GoogleProvider).Records` method returns all records of supported types for all zones. We then visit all the `endpoint.Endpoint`s extracted from those records, and create TXT records for any A or CNAME records, because they satisfy `isManagedRecord` by virtue of their record type.

Reply:
Addressed [here](https://github.com/kubernetes-sigs/external-dns/pull/2811#issuecomment-1174409380).